### PR TITLE
Add autoestimator quick start

### PR DESCRIPTION
**Pull request by yushan111**
This PR added a quick start for `AutoEstimator` with pytorch in COMMON USE CASE

You can review the quick start at 
https://auto-estimator.readthedocs.io/en/latest/doc/Orca/QuickStart/orca-autoestimator-pytorch-quickstart.html

### CODE REVIEW

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nQeViMnCgGaV"
+      },
+      "source": [
+        "After defining your model, you need to define a *Model Creator Function* that returns an instance of your model, and a *Optimizer Creator Function* that returns a PyTorch optimizer. Note that both the *Model Creator Function* and the *Optimizer Creator Function* should takes `config` as input and get the hyper-parameter values from `config`."
```
Comment (shane-huang) [inline docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb:335]: should takes `config` as input -> should take

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "o-SeParPNtK8"
+      },
+      "source": [
+        "### **Step 3: Define Train Dataset**\n",
+        "\n",
+        "You can define the dataset using a *Data Creator Function* that has one parameter of `config` and returns a PyTorch `DataLoader`."
```
Comment (shane-huang) [inline docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb:362]: You can define the dataset -> define the train and test datasets.

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "o-SeParPNtK8"
+      },
+      "source": [
+        "### **Step 3: Define Train Dataset**\n",
```
Comment (shane-huang) [inline docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb:360]: Define Train Dataset -> Define Dataset? (there're train and test datasets definitions)

File: docs/readthedocs/source/doc/Orca/QuickStart/orca-autoestimator-pytorch-quickstart.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+[Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/) is needed to prepare the Python environment for running this example. Please refer to the [install guide](../../UserGuide/python.md) for more details.
+
+```bash
+conda create -n zoo python=3.7 # zoo is conda environment name, you can use any name you like.
+conda activate zoo
+pip install analytics-zoo[ray]
+pip install torch==1.7.1 torchvision==0.8.2
+```
+
+### **Step 1: Init Orca Context**
```
Comment (shane-huang) [inline docs/readthedocs/source/doc/Orca/QuickStart/orca-autoestimator-pytorch-quickstart.md:22]: Would people be confusing what is Orca? Maybe change this title to `Initalization`.

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+        "### **Step 5: Automatically fit and search *with* Orca AutoEstimator**"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "49Rq_fo2O34O"
+      },
+      "source": [
+        "First, Create an AutoEstimator."
```
Comment (shane-huang) [inline docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb:444]: Create -> create

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb [hunk 6]
```diff
@@ -0,0 +1,573 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tFQCQwH0UBF1"
+      },
+      "source": [
+        "Finally, you can get the best learned model and the best hyper-parameters."
+      ]
+    },
```
Comment (shane-huang) [inline docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb:498]: Shall we add a reference to the AutoEstimator API docs? as we did for hp?

**Review by shane-huang — COMMENTED** [review]
Generally LGTM

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb (hunk 3 above)
Comment (yushan111) [reply docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb]: fixed

File: docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb (hunk 6 above)
Comment (yushan111) [reply docs/docs/colab-notebook/orca/quickstart/autoestimator_pytorch_lenet_mnist.ipynb:498]: Sure, I have added the reference after ["First, create an AutoEstimator."](
https://github.com/intel-analytics/analytics-zoo/pull/4179/commits/68803f64ea10aa3da4a254503ce6ceba17e67f96#diff-f4d2a5f2bbb57ce0102b308a6e32ddd05db7c7bd2c6f262684e7dde7d38498d9R444)

**Comment by yushan111** [comment]
http://10.239.176.111:18888/view/ZOO-PR/job/ZOO-PR-Validation/5963/

I will merge this PR and resolve further comments in new PR.

**Pull request merged**